### ISSUE TITLE
Update DataPreprocessing function

### ISSUE THREAD

**Issue by Raminghorbanii** [manuGil/check-in#4]
We need to modify the code for data preprocessing as it now has a major mistake. The normalization should be done based on training data on the test data. However, we are now doing the normalization separately on both train and test sets.
